### ISSUE TITLE
魔术贴软件模拟代码3

### ISSUE THREAD

**Issue by dsakoe** [dsakoe/dsa#14]
hook = self.attached_to

    self.attached_to = None

    hook.attached_to = None

    print(f"Loop {self.loop_id} detached from Hook {hook.hook_id}")

    return True

 

 

# 测试代码

if __name__ == "__main__":

  # 创建钩和环

  hook1 = Hook("H1")

  hook2 = Hook("H2")

  loop1 = Loop("L1")

  loop2 = Loop("L2")

   

  # 尝试连接

  hook1.attach(loop1) # 成功

  hook2.attach(loop1) # 失败，因为loop1已被占用

   

  # 尝试连接另一个

  hook2.attach(loop2) # 成功

   

  # 断开连接

  hook1.detach()   # 成功

  hook2.detach()   # 成功
